penty/pentypes/builtins.py

```diff
@@ -291,6 +291,136 @@ def float_str(self_ty):
 for slot in ('add', 'divmod', 'floordiv', 'mod', 'mul', 'pow', 'sub', 'truediv'):
     _float_attrs['__r{}__'.format(slot)] = _float_attrs['__{}__'.format(slot)]
 
+##
+#
+
+def complex_abs(self_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return complex
+
+def complex_bool(self_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return bool
+
+def complex_divmod(self_ty, other_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    raise TypeError
+
+def complex_complex(self_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return complex
+
+def complex_init(real_ty, imag_ty=None):
+    from penty.penty import Types
+    real_ty = _astype(real_ty)
+
+    if imag_ty is None:
+        if issubclass(real_ty, complex):
+            return complex
+        Types[real_ty]['__float__'](real_ty)
+        return complex
+
+    # interestingly, if the real part is a complex, everything is fine (!)
+    imag_ty = _astype(imag_ty)
+    if issubclass(real_ty, complex):
+        if issubclass(imag_ty, complex):
+            return complex
+        Types[imag_ty]['__float__'](imag_ty)
+        return complex
+
+    Types[real_ty]['__float__'](real_ty)
+    if issubclass(imag_ty, complex):
+        return complex
+    Types[imag_ty]['__float__'](imag_ty)
+    return complex
+
+def complex_eq(self_ty, other_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return bool
+
+def complex_ne(self_ty, other_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    return bool
+
+def complex_int(self_ty):
+    if not issubclass(self_ty, complex):
+        raise TypeError
+    raise TypeError
```

Comment:
Again, the test is useless (until the moment you want to embed additional information in your exception).